jax/experimental/mosaic/gpu/utils.py

```diff
@@ -1115,6 +1124,80 @@ def memref_ptr(memref_arg, memory_space=None):
   )
 
 
+def extern_elementwise_func_apply(func, return_type, arg_list,
+                                  static_arg_idxs=(), pack_elts=False,
+                                  elt_byte_count_idx=None):
+  tidx = thread_idx()
+  wgidx = warpgroup_idx()
+  index = ir.IndexType.get()
+  i64 = ir.IntegerType.get_signless(64)
+  i32 = ir.IntegerType.get_signless(32)
+  ptr_ty = ir.Type.parse("!llvm.ptr")
+  num_elts = None
+  elem_bytewidth = None
+  num_args = len(arg_list)
+  tensor_arg_idxs = set(range(num_args)) - set(static_arg_idxs)
+  wg_thread_offset = arith.muli(wgidx, c(WARPGROUP_SIZE, i32))
+  wg_tidx = arith.subi(tidx, wg_thread_offset)
+  num_cuda_threads = WARPGROUP_SIZE
+
+  arg_list_ptr_int = arg_list.copy()
+  for idx in tensor_arg_idxs:
```

Comment:
Instead of
```python
l2 = l.copy()
for i in ...:
  arg = l2[i]
  ...
  l2[i] = ...
```
it would be much cleaner to write this as:
```python
llvm_args = []
for arg in args:  # Just call the args "args", not "arg_list"
  if ir.MemRefType.isinstance(arg.type):
    arg = ...
  llvm_args.append(arg)
 
```